nilearn/decomposition/base.py

```diff
@@ -0,0 +1,533 @@
+"""
+Base class for decomposition estimators, utilies for masking and reducing group
+data
+"""
+from __future__ import division
+
+import atexit
+import os
+from math import ceil
+from tempfile import mkstemp, mkdtemp
+import warnings
+import numpy as np
+from scipy import linalg
+from sklearn.externals.joblib import Memory, Parallel, delayed
+from sklearn.linear_model import LinearRegression
+from sklearn.utils import check_random_state
+from sklearn.utils.extmath import randomized_svd
+from sklearn.base import BaseEstimator
+from ..input_data import NiftiMapsMasker
+from ..input_data.masker_validation import check_embedded_nifti_masker
+from .._utils.cache_mixin import CacheMixin, cache
+from .._utils.niimg_conversions import check_niimg_4d
+from .._utils.niimg import _safe_get_data
+
+class MaskReducer(BaseEstimator):
```

Comment:
I suggest that we don't support persistence to memmap and use a function instead of this object.